pkg/target/manager.go

```diff
@@ -31,10 +32,28 @@ var (
 	ErrAborted                       = errors.New("operation aborted")
 )
 
+// AutoCompletionsFuncs provides a set of functions that lists suitable values for
+// targeting while taking the currently targeted garden account.
+// In case the manager gets refactored/split up these functions should stay together.
+type AutoCompletionsFuncs interface {
+	// ShootNames returns all shoots for the current target.
+	ShootNames(ctx context.Context) ([]string, error)
+	// SeedNames returns all seeds for the current target.
+	// The current target must at least point to a garden.
```

Comment:
> `// The current target must at least point to a garden.`

This applies all functions except `GardenNames`. Either we remove it here (as we hinted it already in the interface description or add it also for the other functions